src/controller.cpp

```diff
@@ -162,25 +162,33 @@ bool controller::setup_dirs_xdg(const char *env_home, bool silent) {
 	return true;
 }
 
-void controller::setup_dirs(bool silent) {
+void controller::setup_dirs(const char *custom_home, bool silent) {
 	const char * env_home;
-	if (!(env_home = ::getenv("HOME"))) {
-		struct passwd * spw = ::getpwuid(::getuid());
-		if (spw) {
-			env_home = spw->pw_dir;
-		} else {
-			std::cerr << _("Fatal error: couldn't determine home directory!") << std::endl;
-			std::cerr << utils::strprintf(_("Please set the HOME environment variable or add a valid user for UID %u!"), ::getuid()) << std::endl;
-			::exit(EXIT_FAILURE);
+	if(custom_home){
```

Comment:
We're putting a space between a keyword and an opening parenthesis to distinguish them from function calls, like this:
```
if (custom_home) {
```

Please also note the space before the curly brace. Can you apply that style here and in the rest of the patchset?